src/overlays/actors/ovl_En_Col_Man/z_en_col_man.c

```diff
@@ -39,34 +61,220 @@ const ActorInit En_Col_Man_InitVars = {
     (ActorFunc)NULL,
 };
 
-#endif
+static Color_RGBA8 primColor = { 60, 50, 20, 255 };
+static Color_RGBA8 envColor = { 40, 30, 30, 255 };
+
+extern Gfx D_0405AAB0[];
+extern Gfx D_0405F6F0[];
+extern Gfx D_0406AB30[];
+
+void EnColMan_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnColMan* this = THIS;
+
+    Collider_InitAndSetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    this->actor.targetMode = 1;
+    this->scale = (BREG(55) / 1000.0f) + 0.01f;
+
+    switch (this->actor.params) {
+        case EN_COL_MAN_HEART_PIECE:
+        case EN_COL_MAN_RECOVERY_HEART:
+        default:
+            ActorShape_Init(&this->actor.shape, 0.0f, func_800B3FC0, 10.0f);
+            func_80AFDD60(this);
+            break;
+        case EN_COL_MAN_FALLING_ROCK:
+            ActorShape_Init(&this->actor.shape, 0.0f, func_800B3FC0, 10.0f);
+            func_80AFDF60(this);
+            break;
+        case EN_COL_MAN_CUTSCENE_BOMB:
+        case EN_COL_MAN_GAMEPLAY_BOMB:
+            func_80AFE234(this);
+            break;
+    }
+}
+
+void EnColMan_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnColMan* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void func_80AFDD60(EnColMan* this) {
+    if (!(gSaveContext.weekEventReg[56] & 2)) {
+        this->actor.draw = func_80AFE414;
+        this->actor.shape.yOffset = 700.0f;
+        if (this->actor.params == EN_COL_MAN_HEART_PIECE) {
+            this->actor.gravity = -2.0f;
+        }
+        this->unk148 = 0;
+    } else {
+        this->actor.draw = func_80AFE4AC;
+        this->actor.shape.yOffset = 300.0f;
+        this->actor.shape.shadowScale = 5.0f;
+        if (this->actor.params == EN_COL_MAN_HEART_PIECE) {
+            this->actor.gravity = -2.0f;
+        }
+        this->unk148 = 1;
```

Comment:
```suggestion
        this->unk148 = EN_COL_MAN_HEART_PIECE;
    } else {
        this->actor.draw = func_80AFE4AC;
        this->actor.shape.yOffset = 300.0f;
        this->actor.shape.shadowScale = 5.0f;
        if (this->actor.params == EN_COL_MAN_HEART_PIECE) {
            this->actor.gravity = -2.0f;
        }
        this->unk148 = EN_COL_MAN_RECOVERY_HEART;
```
Taking a read through the code, it looks like `this->unk148` stores the type of actor this represents and aligns with the enum you've made below. Thoughts?